cmd/template-resolver/utils/resolver_utils.go

```diff
@@ -72,11 +72,9 @@ func HandleFile(yamlFile string) ([]byte, error) {
 // ProcessTemplate takes a YAML byte array input, unmarshals it to a Policy, ConfigPolicy,
 // or object-templates-raw, processes the templates, and marshals it back to YAML,
 // returning the resulting byte array. Validation is performed along the way, returning
-// an error if any failures are found. It uses the `hubKubeConfigPath`, `hubNS` and `clusterName`
+// an error if any failures are found. It uses the `HubKubeConfigPath`, `HubNamespace` and `ClusterName`
 // to establish a dynamic client with the hub to resolve any hub templates it finds.
-func ProcessTemplate(yamlBytes []byte, hubKubeConfigPath, clusterName, hubNS,
-	objNamespace, objName, saveResourcesPath, saveHubResourcesPath string,
-) ([]byte, error) {
+func ProcessTemplate(yamlBytes []byte, t TemplateResolver) ([]byte, error) {
```

Comment:
Since this is already a "breaking" change (since the function signature changes) I wonder if we should just make it a method on the TemplateResolver?

Reply:
Good point!